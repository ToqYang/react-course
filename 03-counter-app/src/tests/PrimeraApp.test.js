import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas en primera app", () => {
  //   test("Debe mostrar hola soy goku", () => {
  //     const saludo = "hola soy goku";
  //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //     expect(getByText(saludo)).toBeInTheDocument;
  //   });

  test("debe mostrar PrimeraApp correctamente", () => {
    const saludo = "hola soy goku";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar el subtitulo con props", () => {
    const saludo = "hola soy goku";
    const subtitulo = "Soy un sub";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );

    const textoParrafo = wrapper.find("p").text();

    expect(textoParrafo).toBe(subtitulo);
  });
});
