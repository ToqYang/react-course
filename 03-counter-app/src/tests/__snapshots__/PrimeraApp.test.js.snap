// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primera app debe mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    hola soy goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
